### PR TITLE
Add functionality for Batching and using a rewardsHolder

### LINKED ISSUE #2: Add `batch_claim` function and tests

### PR DESCRIPTION
**Pull request by theref**
Fixes #2 #3 #6

### CODE REVIEW

File: contracts/CumulativeMerkleDrop.sol [hunk 1]
```diff
@@ -15,25 +13,39 @@ contract CumulativeMerkleDrop is Ownable, ICumulativeMerkleDrop {
...
+        bytes32[] proof;
+    }
 
-    constructor(address token_) {
+    event RewardsHolderUpdated(address oldRewardsHolder, address newRewardsHolder);
+
+    constructor(address token_, address rewardsHolder_) {
         token = token_;
+        rewardsHolder = rewardsHolder_;
     }
```
Comment (vzotova) [inline contracts/CumulativeMerkleDrop.sol:31]: I know that is not necessary but would be great to add check for token contract here, something like `token.totalSupply > 0`. And that is `rewardsHolder`  is not zero address

File: contracts/CumulativeMerkleDrop.sol [hunk 2]
```diff
@@ -15,25 +13,39 @@ contract CumulativeMerkleDrop is Ownable, ICumulativeMerkleDrop {
...
 
     function setMerkleRoot(bytes32 merkleRoot_) external override onlyOwner {
         emit MerkelRootUpdated(merkleRoot, merkleRoot_);
         merkleRoot = merkleRoot_;
     }
 
+    function setRewardsHolder(address rewardsHolder_) external onlyOwner {
+        emit RewardsHolderUpdated(rewardsHolder, rewardsHolder_);
+        rewardsHolder = rewardsHolder_;
```
Comment (vzotova) [inline contracts/CumulativeMerkleDrop.sol:40]: Maybe add address check here too? maybe not only for zero address but that some tokens belong to that address or even allowance? wdyt?

**Review by vzotova — APPROVED** [review]

File: contracts/CumulativeMerkleDrop.sol (hunk 2 above)
Comment (theref) [reply contracts/CumulativeMerkleDrop.sol:40]: i think checking that it's not 0 address is a good idea. I think checking for the allowance/balance might be premature - it's possible that those would only be sorted afterwards.

File: contracts/CumulativeMerkleDrop.sol (hunk 1 above)
Comment (theref) [reply contracts/CumulativeMerkleDrop.sol:31]: good idea

**Review by cygnusv — APPROVED** [review]
LGTM!

**Pull request merged**